pkg/controllers/nodeclaim/lifecycle/initialization.go

```
@@ -54,38 +57,73 @@ func (i *Initialization) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim)
+	nodeClaim.StatusConditions().SetTrue(v1.ConditionTypeInitialized)
+	return reconcile.Result{}, nil
+}
+
+func updateNodeClaimInitializedUnknown(node *corev1.Node, nodeClaim *v1.NodeClaim) bool {
 	if nodeutils.GetCondition(node, corev1.NodeReady).Status != corev1.ConditionTrue {
 		nodeClaim.StatusConditions().SetUnknownWithReason(v1.ConditionTypeInitialized, "NodeNotReady", "Node status is NotReady")
-		return reconcile.Result{}, nil
+		return true
 	}
 	if taint, ok := StartupTaintsRemoved(node, nodeClaim); !ok {
 		nodeClaim.StatusConditions().SetUnknownWithReason(v1.ConditionTypeInitialized, "StartupTaintsExist", fmt.Sprintf("StartupTaint %q still exists", formatTaint(taint)))
-		return reconcile.Result{}, nil
+		return true
 	}
 	if taint, ok := KnownEphemeralTaintsRemoved(node); !ok {
 		nodeClaim.StatusConditions().SetUnknownWithReason(v1.ConditionTypeInitialized, "KnownEphemeralTaintsExist", fmt.Sprintf("KnownEphemeralTaint %q still exists", formatTaint(taint)))
-		return reconcile.Result{}, nil
+		return true
 	}
 	if name, ok := RequestedResourcesRegistered(node, nodeClaim); !ok {
 		nodeClaim.StatusConditions().SetUnknownWithReason(v1.ConditionTypeInitialized, "ResourceNotRegistered", fmt.Sprintf("Resource %q was requested but not registered", name))
-		return reconcile.Result{}, nil
+		return true
 	}
-	stored := node.DeepCopy()
-	node.Labels = lo.Assign(node.Labels, map[string]string{v1.NodeInitializedLabelKey: "true"})
-	if !equality.Semantic.DeepEqual(stored, node) {
-		if err = i.kubeClient.Patch(ctx, node, client.MergeFrom(stored)); err != nil {
-			return reconcile.Result{}, err
+	return false
+}
+
+func (i *Initialization) updateNodePoolRegistrationHealth(ctx context.Context, nodeClaim *v1.NodeClaim) (bool, error) {
+	nodePool := &v1.NodePool{}
+	if err := i.kubeClient.Get(ctx, types.NamespacedName{Name: nodeClaim.Labels[v1.NodePoolLabelKey]}, nodePool); err != nil {
+		return false, err
```

This should probably just return the error and be handled in the reconcile like we do in other reconcile loops.